src/NodeChanger/MethodNameChanger.php

```diff
@@ -3,20 +3,21 @@
 namespace Rector\NodeChanger;
 
 use PhpParser\Node;
-use PhpParser\Node\Expr\MethodCall;
 use PhpParser\Node\Identifier;
 
 final class MethodNameChanger
 {
     /**
-     * @param string[] $renameMethodMap
+     * @param string|string[] $newMethodName
      */
-    public function renameNode(MethodCall $node, array $renameMethodMap): ?Node
+    public function renameNode(Node $node, $newMethodName): void
     {
-        $oldNodeMethodName = $node->name->toString();
+        if (is_array($newMethodName)) {
```

Comment:
`if (is_<something>)` is code smell that suggest method decoupling

Reply:
The other method could be `renameNodeWithMap(Node $node, array $renameMethodMap)`

Reply:
Yep, I though in `renameToNode` in case of a `string` given and `renameNodeWithMap` with an `array` was passed!

Reply:
That is ambiguous behavior for outside scope.

Usually I'd recommend you *Clean Code* or [Principles of Package Design](https://leanpub.com/principles-of-package-design), but you see this short version ;):

- https://github.com/jupeter/clean-code-php#avoid-conditionals
- https://github.com/jupeter/clean-code-php#dont-use-flags-as-function-parameters
-

Reply:
@TomasVotruba Couldn't have a better teacher. I'm learning a lot working with you!